upload/extension/opencart/admin/model/payment/opayo.php

```diff
@@ -0,0 +1,406 @@
+		$qry = $this->db->query("SELECT * FROM `" . DB_PREFIX . "opayo_order` WHERE `order_id` = '" . (int)$order_id . "' LIMIT 1");
+
+		if ($qry->num_rows) {
+			$order = $qry->row;
+			$order['transactions'] = $this->getOrderTransactions($order['opayo_order_id']);
+
+			return $order;
+		} else {
+			return [];
+		}
+	}
+
+	/**
+	 * Get Order Transactions
+	 *
+	 * @param int $opayo_order_id
+	 *
+	 * @return array<int, array<string, mixed>>
+	 */
+	private function getOrderTransactions(int $opayo_order_id): array {
+		$query = $this->db->query("SELECT * FROM `" . DB_PREFIX . "opayo_order_transaction` WHERE `opayo_order_id` = '" . (int)$opayo_order_id . "'");
+
+		if ($query->num_rows) {
+			return $query->rows;
+		} else {
+			return [];
+		}
+	}
+
+	/**
+	 * Add Order Transaction
+	 *
+	 * @param int    $opayo_order_id
+	 * @param string $type
+	 * @param float  $total
+	 *
+	 * @return void
+	 */
+	public function addOrderTransaction(int $opayo_order_id, string $type, float $total): void {
+		$this->db->query("INSERT INTO `" . DB_PREFIX . "opayo_order_transaction` SET `opayo_order_id` = '" . (int)$opayo_order_id . "', `date_added` = now(), `type` = '" . $this->db->escape($type) . "', `amount` = '" . (float)$total . "'");
+	}
+
+	/**
+	 * Get Total Released
+	 *
+	 * @param int $opayo_order_id
+	 *
+	 * @return float
+	 */
+	public function getTotalReleased(int $opayo_order_id): float {
+		$query = $this->db->query("SELECT SUM(`amount`) AS `total` FROM `" . DB_PREFIX . "opayo_order_transaction` WHERE `opayo_order_id` = '" . (int)$opayo_order_id . "' AND (`type` = 'payment' OR `type` = 'rebate')");
+
+		return (float)$query->row['total'];
+	}
+
+	/**
+	 * Get Total Rebated
+	 *
+	 * @param int $opayo_order_id
+	 *
+	 * @return float
+	 */
+	public function getTotalRebated(int $opayo_order_id): float {
+		$query = $this->db->query("SELECT SUM(`amount`) AS `total` FROM `" . DB_PREFIX . "opayo_order_transaction` WHERE `opayo_order_id` = '" . (int)$opayo_order_id . "' AND 'rebate'");
+
+		return (float)$query->row['total'];
+	}
+
+	/**
+	 * Send Curl
+	 *
+	 * @param string       $url
+	 * @param array<mixed> $payment_data
+	 *
+	 * @return array<array, string>
+	 */
+	public function sendCurl(string $url, array $payment_data): array {
+		$curl = curl_init($url);
+
+		curl_setopt($curl, CURLOPT_PORT, 443);
+		curl_setopt($curl, CURLOPT_HEADER, 0);
+		curl_setopt($curl, CURLOPT_SSL_VERIFYPEER, 0);
+		curl_setopt($curl, CURLOPT_RETURNTRANSFER, 1);
+		curl_setopt($curl, CURLOPT_FOLLOWLOCATION, false);
+		curl_setopt($curl, CURLOPT_FORBID_REUSE, 1);
+		curl_setopt($curl, CURLOPT_FRESH_CONNECT, 1);
+		curl_setopt($curl, CURLOPT_POST, 1);
+		curl_setopt($curl, CURLOPT_POSTFIELDS, http_build_query($payment_data));
+
+		$response = curl_exec($curl);
+
+		curl_close($curl);
+
+		$response_info = explode(chr(10), $response);
+
+		foreach ($response_info as $i => $string) {
+			if (strpos($string, '=') && isset($i)) {
```

Comment:
An array key cannot be unset or `null`
```suggestion
			if (strpos($string, '=')) {
```
Alternatively you can check if it's falsy or an empty string.

Reply:
Yeah, that's what I got on my end. Putting it in now.